### DISCUSSION TITLE
NOT IN query with multiple IN values using Supabase?

### DISCUSSION THREAD

**Discussion by ShubhamVerma1811** [supabase/supabase#2055] [Questions] [answered]
Hey,
How can I replicate the SQL query for NOT IN?
`SELECT * FROM t WHERE t.name LIKE %a% AND t.name NOT IN ('x','y','x')`

Looking at the docs I found `not()` and `neq()` but not sure how it would work for multiple `IN` values.
Thank you for your time and help.

**Comment by steve-chavez** [comment, marked as answer]
Hey @ShubhamVerma1811,

You can use the [in](https://supabase.io/docs/reference/javascript/in) and [not](https://supabase.io/docs/reference/javascript/not) filters for that.

```js
const { data, error } = await supabase
  .from('t')
  .select('*')
  .like('name', '%a%')
  .not('name', 'in', ['x', 'y', 'z']))
```

**Reply by ShubhamVerma1811** [reply]
Thank you for that!

**Reply by ShubhamVerma1811** [reply]
Hey @steve-chavez ,
I tried this today and it doesn't seem to work. It returns null. What am I doing wrong here?
```js
 const { data, error } = await supabase
    .from("tags")
    .select("name")
    .not("tid", "in", [1, 2])
    // and also    .not("name", "in", ['CSS'])
```

This the table schema
```json
[
  {
    "tid": 1,
    "color": "#61dafb",
    "name": "React",
    "description": "JS Framework made by Facebook"
  },
  {
    "tid": 2,
    "color": "#1e6fb0",
    "name": "CSS",
    "description": "CSS styles the web"
  },
  {
    "tid": 3,
    "color": "#f6dd1e",
    "name": "JavaScript",
    "description": "Lang of the web"
  }
]
```

**Reply by steve-chavez** [reply]
@ShubhamVerma1811 Sorry, there seems to be a bug in [postgrest-js](https://github.com/supabase/postgrest-js). For now you can do it like this:

(The in value must be a string surrounded by parens) 

```js
const { data, error } = await supabase
    .from("tags")
    .select("name")
    .not("tid", "in", "(1, 2)")
```

**Reply by ShubhamVerma1811** [reply]
Yes this approach seems to work and I liked it compared to first way.
Because now I dont need to extract single string from something like `"react,css"` to `"react","css"`
rather just pass the whole string as  `"react,css"`

**Reply by ian-speckart** [reply]
`.not("tid", "in", "(1, 2, 3)")`

It's currently filtering out only the first param for me (eg, only '1' is skipped, the rest are included in the result).
Do you also have this issue?

**Reply by osmanmesutozcan** [reply]
Yeah. It almost wrecked our database....

Turns out you should NOT have spaces in between. And we had that on an update query...

Only reason we didn't have a catasrophic issue was because the first item in the array was the most important one.
